### DISCUSSION TITLE
Synology NAS Support

### DISCUSSION THREAD

**Discussion by Kovah** [Kovah/LinkAce#371] [Ideas & Feature Requests]
This topic continues the discussion of a post from the old community forum: "Synology NAS Support" by adam
A backup of the full conversation can be viewed on the [Internet Archive](https://web.archive.org/web/20220130214926/https://community.linkace.org/t/synology-nas-support/71).

---

> Is your feature request related to a problem? Please describe.
> It would be great to have support for installation onto a NAS device, as I know there’s a very large community of Synology users utilizing Synology’s Docker manager. This Docker manager does not allow command-line setup and everything is preconfigured with environment variables. I believe LinkAce is the best solution out there for this problem, but it cannot be easily installed on a NAS and therefore I have to consider alternative solutions.
> 
> Perhaps it’s just a documentation issue, but Volume and all Environmental Variables need to be defined before launching Docker container.
> 
> Describe the solution you’d like
> Support for installation onto Synology NAS Docker manager. Does not have access to command-line installation. Everything pre-configured with environmental variables.
> 
> Describe alternatives you’ve considered
> Other bookmark managers.

**Comment by soundneedle** [comment]
I wasn't able to get this to work on my synology either. Get to add user webpage and is unable to add getting error below. I'm assuming this isn't related to synology mariadb since the db account was able to create the linkace tables. No idea why I can't now add a user though.

`SQLSTATE[HY000] [1045] Access denied for user 'linkace'@'172.17.0.9' (using password: YES) (SQL: select count(*) as aggregate from `users` where `email` = xxxx@xxx.com)`

**Comment by martadinata666** [comment]
the problem is how the db charset as default is latin and linkace need utf-8, on compose need modify mysql command
```
command: mysqld --character-set-server=utf8mb4 --collation-server=utf8mb4_bin
```
I dont use syno, no idk how to change the default command for it. But the root issue is this charset.

**Reply by Kovah** [reply]
It is possible to set the encoding on a database table level when you create it, independent from MySQL server itself. The command is only there in case LinkAce is the sole user of the MySQL server.

```
CREATE DATABASE name_of_new_database CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
```
